src/ios/VLCPlugin.m

```diff
@@ -325,48 +373,70 @@ - (void)playremotefile:(CDVInvokedUrlCommand*)command {
     }
     
-    [self vlc_sendPluginResult:pluginResult callbackId:command.callbackId];
+    [self vlc_sendPluginResult:pluginResult callbackId:self.callbackId];
 }
 
 - (void)pause:(CDVInvokedUrlCommand*)command {
+    [self pauseWithCallbackId:command.callbackId];
+}
+
+- (void)stop:(CDVInvokedUrlCommand*)command {
+    [self stopPlayback];
+}
+
+- (void)setaudioinfo:(CDVInvokedUrlCommand*)command{
+    NSDictionary  * info = [command.arguments  objectAtIndex:0];
+    [self vlc_setlockscreenmetadata:info refreshLockScreen:true];
+    
+    if ([self.audio isKindOfClass:[NYPRStream class]] && [info objectForKey:@"svg_logo"]) {
+        NYPRStream *stream = [NYPRStream createStream:[[NYPRPersistentStack sharedInstance] managedObjectContext]];
+        [stream loadFromDictionary:info];
+        self.audio = stream;
+        self.audio.state = self.mediaPlayer.state;
+        self.audioPlayerViewController.audio = self.audio;
+    }
+}
+
+- (void)vlc_setlockscreenmetadata:(NSDictionary*)metadata refreshLockScreen:(BOOL)refreshLockScreen {
+    self.lockScreenCache = [NSDictionary dictionaryWithDictionary:metadata];
+    if(refreshLockScreen){
+        [self vlc_setNowPlayingInfo:self.lockScreenCache retrieveLockscreenArt:YES];
+    }
+}
+
+#pragma mark Audio playback controls
+
+- (void)pauseWithCallbackId:(NSString *)callbackId {
     DDLogInfo (@"VLC Plugin pausing playback");
     if (self.mediaPlayer.isPlaying) {
         [self.mediaPlayer pause];
+        
+        if (self.audio) {
+            self.audio.state = NYPRAudioStatePaused;
+        }
     }
     
     CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
-    [self vlc_sendPluginResult:pluginResult callbackId:command.callbackId];
+    [self vlc_sendPluginResult:pluginResult callbackId:callbackId];
 }
 
-- (void)seek:(CDVInvokedUrlCommand*)command {
-    CDVPluginResult* pluginResult = nil;
-    NSInteger interval = [[command.arguments objectAtIndex:0] integerValue];
-    
-    if ([self.mediaPlayer isSeekable]){
-        DDLogInfo (@"VLC Plugin seeking to interval (%ld)", (long)interval );
-        if (interval>0){
-            [self.mediaPlayer jumpForward:((int)interval/1000)];
-        }else{
-            [self.mediaPlayer jumpBackward:(-1*(int)interval/1000)];
-        }
-        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
-    } else {
-        DDLogInfo (@"VLC Plugin current audio is not seekable" );
-        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"audio not seekable"];
+- (void)stopPlayback {
+    DDLogInfo (@"VLC Plugin stopping playback.");
+    if (self.mediaPlayer.isPlaying) {
+        [self.mediaPlayer stop];
+        self.audio.state = NYPRAudioStateStopped;
+        [self vlc_onAudioStreamUpdate:MEDIA_STOPPED description:[self vlc_convertAudioStateToString:MEDIA_STOPPED]];
```

Comment:
Same comment as above -- would relying on mediaPlayer.stop be sufficient for changing the state?